include/boost/hash2/xxh3.hpp

```diff
@@ -535,11 +535,16 @@ class xxh3_128
         seed_ = seed;
     }
 
-private: // supporting constructor for the static factory functions
+private:
 
-    BOOST_CXX14_CONSTEXPR xxh3_128( std::uint64_t seed, unsigned char const* p, std::size_t n ): seed_( seed )
+    // supporting constructor for the static factory functions
+    // we use the trailing `with_secret` parameter so that the constructors can use
+    // prvalue'd return expressions
+    // versions of msvc seem to dislike calling private constructors during constant evaluation
+    // if they yield an lvalue, i.e. something of the of the form: `xxh3_128 r;`
+    BOOST_CXX14_CONSTEXPR xxh3_128( std::uint64_t seed, unsigned char const* p, std::size_t n, bool with_secret ): seed_( seed )
     {
-        with_secret_ = true;
+        with_secret_ = with_secret;
```

Comment:
I suppose this can now go in the member initializer list, for aesthetic reasons if nothing else.